### ISSUE TITLE
Bodies go crazy when two motors are trying to push against one another

### ISSUE THREAD

**Issue by josephrocca** [schteppe/cannon.js#228]
I've been noticing a bit of crazy physics happening in my program lately and I think I have narrowed it down to this: whenever two motors are opposing each other and the motor toques are above a certain threshold, things get really crazy. So take the following situation:
![cannonissue](https://cloud.githubusercontent.com/assets/1167575/9921783/2b1d1824-5d14-11e5-9b54-85fd3124a301.jpg)
Bodies B and C are attached by a hinge to body A, and their motor speeds are set so they push into one another. Setting a small `maxMotorForce` doesn't cause any problems, but if any reasonable force is required, then the bodies start flipping around - it looks like they're being pulled around and spun by some invisible force.

I've tried increasing the solver iterations up to 30 - and it definitely helps, but there is still some wierd forces going on. I think the wierdness tend to happen when there's enough force to cause the bodies to overlap one another a bit.

Is this a bug, or just an intrinsic thing about physics engines that I'll have to work around if I need large motor forces? I'm guessing it's the latter, so in that case, are there any tips to dampen the craziness, other than playing with `solverIterations`?

Here's another clue I just found:
![cannonissue2](https://cloud.githubusercontent.com/assets/1167575/9922209/62c12e10-5d18-11e5-8b45-8655d7b1ca72.jpg)
This is with `solverIterations` at 30 and `maxMotorForce` at around 200. The bodies are also jittering quite a bit.

Thanks!

**Comment by schteppe** [comment]
Hi,
The hinge motor constraint tries to make the relative rotational velocity of the attached bodies equal a given number. But since they get blocked, I guess the velocity has to go somewhere else... Maybe this would work better with a new motor constraint type that used angles instead of angular velocities, not sure.
The max force, the number of iterations, time step, etc, are what you have to play with.

I wish I could help, but I don't know how to solve it :P

**Comment by josephrocca** [comment]
Ah, okay, no worries! I haven't played with the time step yet, so I'll give that a try. Thanks :)

**Comment by josephrocca** [comment]
Okay, so I've played around with this quite a bit now, and the problem looks more serious than I initially assumed. I've tried adjusting the time step, iterations and max force to no avail.

However, a sneaky solution that seems to be working for me and may help others who come across this issue is: whenever the two bodies connected by the hinge are in contact with one another, inhibit the motor speed by about 90%.

This allows the motors to be at their normal speed when they're going through their normal range of motion, but slow right down when they are about to cause 'self-collision quantum wierdness'.

If anyone has a better/different way to handle this I'd love to hear about it! :)
